### PR TITLE
coronavirus API

### PR DESCRIPTION
**Pull request by cul3bro**
This pull request adds the functionality of the coronavirus API

### CODE REVIEW

File: src/main/java/es/unizar/webeng/hello/coronaAPI.java [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.ObjectMapper;
+
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.client.RestTemplate;
+
+/**
+ * This class is used to zip all about HTTP protocol and GET request
+ */
+public class coronaAPI {
```
Comment (fjlopez) [inline src/main/java/es/unizar/webeng/hello/coronaAPI.java:12]: This class should be named `CoronaAPI` and  annotated as a @Service and injected into the HelloController. A public method named e.g.  `coronaAPI` should return an `Optional<CoronaDTO>`. This optional must be returned empty if any exception is detected.

File: src/main/java/es/unizar/webeng/hello/HelloController.java [hunk 2]
```diff
@@ -149,6 +149,18 @@ public String rollTheDice(Map<String, Object> model) throws JsonProcessingExcept
                 model.put("temperature", "APIDoesNotRespond");
             }
         }
+
+        // Getting coronavirus information
+        coronaAPI corona = new coronaAPI();
```
Comment (fjlopez) [inline src/main/java/es/unizar/webeng/hello/HelloController.java:154]: This should be a call to an injected service. The return must be an `Optional<CoronaDTO>`. The semantics of `Optional` allows a solution similar to the current but standard.

**Review by fjlopez — CHANGES_REQUESTED** [review]
Some key changes are requested. They are related to a better use of the Spring Technology and Java features.

File: src/main/java/es/unizar/webeng/hello/HelloController.java (hunk 2 above)
Comment (cul3bro) [reply src/main/java/es/unizar/webeng/hello/HelloController.java:154]: I can't implement the changes you mention ...

File: src/main/java/es/unizar/webeng/hello/coronaAPI.java (hunk 1 above)
Comment (cul3bro) [reply src/main/java/es/unizar/webeng/hello/coronaAPI.java:12]: I can't implement the changes you mention ...

**Review by fjlopez — APPROVED** [review]

**Pull request merged**